crates/solver/src/settlement_post_processing/optimize_score.rs

```diff
@@ -0,0 +1,49 @@
+use {
+    super::SettlementSimulating,
+    crate::{
+        objective_value::Inputs,
+        settlement::Settlement,
+        solver::score_computation::ScoreCalculator,
+    },
+    gas_estimation::GasPrice1559,
+    num::{BigRational, FromPrimitive},
+    shared::{external_prices::ExternalPrices, http_solver::model::Score},
+};
+
+pub async fn optimize_score(
+    settlement: Settlement,
+    settlement_simulator: &impl SettlementSimulating,
+    score_calculator: &ScoreCalculator,
+    gas_price: GasPrice1559,
+    prices: &ExternalPrices,
+) -> Settlement {
+    if settlement.score.is_some() {
+        return settlement;
+    }
+
+    let gas_amount = match settlement_simulator
+        .settlement_would_succeed(settlement.clone())
+        .await
+    {
+        Ok(gas_amount) => gas_amount * 9 / 10, /* multiply with 0.9 to get more realistic gas */
+        // amount
```

Comment:
Why would is 90% a more realistic gas amount here? Since we actually simulate the tx instead of using hardcoded gas estimates for each trade or interaction shouldn't this value be 100% correct?

Reply:
`eth_estimateGas` returns around 10-15% more gas than actually needed (because it's an _estimation_ function). The only way to calculate the gas exactly is to use Tenderly but IMO eth_estimateGas reduced by 10% already gives pretty good values. (Tenderly calls would take more time and increase our usage by 5-15 calls per auction).

Reply:
But I do agree additional explanation should be added to the comment.

Reply:
> returns around 10-15% more gas than actually needed (because it's an estimation function).

It is also a product of needing it to be the total gas before refunds (an EVM requirement) so it will always be an upper bound on gas spent and not the actual gas spent.

Reply:
Added comment.